src/models/architectures/cnns/architectures/competitive_multiscale_cnn.py

```diff
@@ -0,0 +1,74 @@
+import torch
+import torch.nn as nn
+
+
+class MaxOut(nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, inputs):
+        cur_max = inputs[0]
+        for inp in inputs[1:]:
```

Comment:
I think that `input` and `current_max` names are better (I personally don't like abbreviations, because they can lead to code misunderstanding).

Reply:
Name `input` overrides built-in python function - which I tried to avoid.
I wonder if the name `cur_max` isn't just right for such small scope (only 4-liner function).

Reply:
As `input` name overriding is indeed bad, `current_max` makes it just no-brainer name and in general makes code reading faster.